src/variable.cpp

```diff
@@ -593,6 +606,15 @@ class ViewModel
   getReshaped(const Dimensions &dims) const override {
     return {m_model, dims};
   }
+  VariableView<value_type> getReshaped(const Dimensions &dims) override {
+    requireMutable();
+    if constexpr (std::is_const<typename T::value_type>::value) {
+      static_cast<void>(dims);
```

Comment:
It is necessary to avoid compiler warnings (unused arg). I was surprised as well: with `if constexpr` you get the warning even if it is used in the other branch.

Reply:
Could just eliminate the named variable from the signature. i.e. `(const Dimensions &)`

Reply:
No, it is required in the other branch.